swift/example_code/s3/basics/Sources/ServiceHandler/ServiceHandler.swift

```diff
@@ -207,23 +274,33 @@ public class ServiceHandler {
     ///            one file contained in the bucket.
     // snippet-start:[s3.swift.basics.handler.listbucketfiles]
     public func listBucketFiles(bucket: String) async throws -> [String] {
-        let input = ListObjectsV2Input(
-            bucket: bucket
-        )
-        let output = try await client.listObjectsV2(input: input)
-        var names: [String] = []
-
-        guard let objList = output.contents else {
-            return []
-        }
-
-        for obj in objList {
-            if let objName = obj.key {
-                names.append(objName)
+        do {
+            let input = ListObjectsV2Input(
+                bucket: bucket
+            )
+            let output = client.listObjectsV2Paginated(input: input)
+            var names: [String] = []
+            
+            for try await page in output {
+                guard let objList = page.contents else {
+                    print("ERROR: listObjectsV2Paginated returned nil contents.")
+                    continue
+                }
+                
+                for obj in objList {
+                    if let objName = obj.key {
+                        names.append(objName)
+                    }
+                }
             }
```

Comment:
Add a comment explaining what this is doing. Would be especially handy since pagination isn't yet well-covered in the developer guide (which I hadn't realized until now and am filing myself a ticket for it).

Reply:
done